code/parse/parselo.cpp

```diff
@@ -2327,11 +2327,156 @@ void coerce_to_utf8(SCP_string &buffer, const char *str)
 	Warning(LOCATION, "Truncating non-UTF-8 string '%s' to '%s'!\n", str, buffer.c_str());
 }
 
+//Converts a string in ISO 8859-1 to the currently used encoding
+//Only legal for strings with correct ISO 8859-1 encoding
+SCP_string iso8859_1_to_current(const char* str)
```

Comment:
This looks very similar to `coerce_to_utf8`.  Why can't you use that function?

Reply:
Coerce_to_utf8 has two distinct differences:
first, it _guesses_ whether the source encoding is latin1 based on heuristics, and only if it thinks this is the case will it actually convert. As opposed to that, the iso8859_1_to_current function, will assume it gets iso8859-1 encoding and work from that, even if the heuristic may fail.
Secondly, and more importantly, the function converts it's input to _the current_ encoding. Which might be utf-8, but it might also be iso8859-1, as opposed to the coerce function which will always output utf-8.
But code reuse could probably be reduced by combining both to_current functions to a single conversion function that operates with more parameters, and then adapting the coerce function to perform it's checks and then call upon said conversion function.

Reply:
> But code reuse could probably be reduced by combining both to_current functions to a single conversion function that operates with more parameters, and then adapting the coerce function to perform it's checks and then call upon said conversion function.

Yes, that sound like the best approach.